ECIP 1037: Simple non-interactive URI Scheme

realcodywburns opened the pull request:
Abstract

This ECIP proposes a URI scheme for interacting with ethereum contract and accounts in a standard format.

Motivation

The purpose of this URI scheme is to enable users to easily make payments and interact with smart contracts by simply clicking links on webpages or scanning QR Codes.

realcodywburns committed:
Init

realcodywburns committed:
add bytecode explanition

realcodywburns commented:
Changing `chainid` to `id` for brevity. Addational space can also be saved by using iban address vs account hex.

realcodywburns committed:
Update ECIP-1037.mediawiki

Change `chainid` to `id` to save byte space

drd34d commented:
sounds like an awesome idea to make interaction stupidily simple!

realcodywburns committed:
Update ECIP-1037.mediawiki

Formatting

elaineo commented on ECIPs/ECIP-1037.mediawiki:
```diff
@@ -0,0 +1,196 @@
+=== Examples ===
+
+Just the address:
+ ethereum:0xCe5ED529977b08f87CBc207ebC216859820461eE&id=61
```
How about `chainId` instead of just `id`? to avoid confusion with `networkId`

elaineo commented:
@realcodywburns actually i think it should remain `chainid` to avoid confusion with network id.

realcodywburns committed:
Update ECIP-1037.mediawiki

Add reference to bytecode

realcodywburns commented:
What about `v` or `cid`?

realcodywburns committed:
Update ECIP-1037.mediawiki

Chain id is a base 10 number, not base 16

Dexaran commented:
@realcodywburns 

```js
 function getETC(){
    require(block.number-payoutDrip >= payoutRate);
    msg.sender.transfer(dropValue);   
    }
```

>
>====Example====
>
>Contract C is at 0xdeadbeef... and has the following psudo function:
>
>Contract C requires no inputs and only requires enough gas to preform its functions. The byte code for >this function can be derived using `web3` as:
> 
>web3.sha3("getETC()");
>
>which yeilds:
>
>0x023b88685d10d29e0bf0563e4ab1b9d8fc8333c166705be5538c4b079cdd9af0
>
>The information that will be passed in the byte code field would then be:
>
>code=0x023b88685d

I think that if you need to pass the function signature then it should be `0x023b8868` but not 0x023b8868~~5d~~ .

realcodywburns committed:
Update ECIP-1037.mediawiki

Updated mock function code to correct format of only left most 4 bytes
Rm `5d`

realcodywburns commented:
I updated, it should only be the left most 4 bytes

realcodywburns committed:
Update ECIP-1037.mediawiki

added rawtx mapping to remove confusion
added nonce and gaslimit
added ENS table

realcodywburns commented:
https://github.com/ethereum/EIPs/issues/67

elaineo commented:
@realcodywburns i still prefer chainId for consistency, but it's not a dealbreaker or anything :)

realcodywburns commented:
Closing, this repo is depreciated . To be reopened on http://github.com/ethereumclassic/ECIPs
